.circleci/gitChangedLibs.rb

```diff
@@ -0,0 +1,28 @@
+require 'json'
+require 'set'
+
+$GITPRAPI = "https://api.github.com/repos/%s/SalesforceMobileSDK-android/pulls/%s/files"
+$libs = ["SalesforceAnalytics", "SalesforceHybridSDK", "SalesforceReact", "SalesforceSDK", "SmartStore", "SmartSync"]
+
+prFilesAPI = $GITPRAPI % [ENV["CIRCLE_PROJECT_USERNAME"], ENV["CIRCLE_PR_NUMBER"]]
+pullfiles = `#{"curl %s" % [prFilesAPI]}`
+prfiles = JSON.parse(pullfiles)
+
+libs = Set.new
+for prfile in prfiles
+  path = prfile["filename"]
+  for lib in $libs
+    if path.include? lib
+      libs = libs.add(lib)
+    end
+  end
+end
+
+libs_ar = libs.to_a()
+if !libs_ar.empty?() && libs_ar.include?("SalesforceSDK")
```

Comment:
If files are chanced in Core, run all tests.

Reply:
SalesforceSDK depends on SalesforceAnalytics (see https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/master/libs/SalesforceSDK/build.gradle#L17).
So line 22, we should test for SalesforceAnalytics.
We should have another elsif for SalesforceSDKCore that would push SmartStore, SmartSync, SalesforceHybridSDK, SalesforceReact.

Reply:
We could simplify the code:

```ruby
libsTopoSorted = ["SalesforceAnalytics", "SalesforceSDK", "SmartStore", "SmartSync", "SalesforceHybridSDK", "SalesforceReact"] 
libsModified = libs.to_a()
libsToTest = libsTopoSorted.slice(libsModified.map{|l| libsTopoSorted.find_index(l)}.min() .. libsTopoSorted.length)
```
libsTopoSorted is topologically sorted - x before y means x does not depend on y.
It looks for the modified library that has the smallest index in libsTopoSorted.
Then tests that library and all the ones that follow in libsTopoSorted.